In [10]:
import pandas as pd
import numpy as np

In [11]:
df = pd.read_csv('telco_customer_churn.csv')

In [12]:
#df.customerID.nunique()

df

#berdasarkan nunique jumlah customer berdasarkan customer ID adalah 7043 yg berarti tidak ada customer yg memiliki ID ganda

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7043 rows x 21 columns]

### Welcome to Day 6 Homework. Please answer the questions below!

1. Buanglah baris dengan `tenure` samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan `tenure` TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

2. Bagi customer yang ***TIDAK MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa ***rata-rata*** dari `tenure` mereka?

3. Bagi customer yang ***MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa maksimal `MonthlyCharges` yang tercata?

4. Berapa banyak customer yang ***TIDAK MEMILIKI PARTNER*** namun ***MEMILIKI DEPENDENTS***?

5. Berapa rata-rata ***TOTAL CHARGES*** dari customer ***NON-SENIOR CITIZEN*** (Senior Citizen = 0) yang memiliki ***PHONE SERVICE*** dan memiliki ***INTERNET SERVICE FIBER OPTIC***?

#### Silakan menuliskan kode untuk menjawab 5 pertanyaan di atas...

In [13]:
#soal 1

df_edit = df.copy()

con_1 = df_edit['tenure'] == 0
con_2 = df_edit['tenure'] != 0 #bisa juga menggunakan >0

#df[con_1] #untuk melihat column masing-masing tenure
#df[con_2]

df_edit = df_edit[con_2]
df_edit

#df_edit.info()

#dari 7043 data tersisa menjadi 7032 data yg berarti tenure yg memiliki nilai 0 berjumlah 11

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7032 rows x 21 columns]

In [14]:
#soal 2

con_part_1 = df_edit['Partner'] == 'No'
con_dept_1 = df_edit['Dependents'] == 'No'
con_ten = df_edit['tenure']

filtered_df = df_edit[con_part_1 & con_dept_1 & con_ten]

tabel_baru_1 =  filtered_df[['Partner', 'Dependents', 'tenure']] 

#tabel_baru_1

tabel_baru_1['tenure'].mean()

20.374376039933445

In [15]:
#soal 3

con_part_2 = df_edit['Partner'] == 'Yes'
con_MonthPay = df_edit['MonthlyCharges']

new_con_df = df_edit[con_part_2 & con_dept_1 & con_MonthPay]

tabel_baru_2 = new_con_df[['Partner', 'Dependents', 'MonthlyCharges']]
#tabel_baru_2

#tabel_baru_2.applymap(np.isreal)

tabel_baru_2['MonthlyCharges'].max()

118.6

In [16]:
#soal 4

con_dept_2 = df_edit['Dependents'] == 'Yes'

aselole = df_edit[con_part_1 & con_dept_2]

tabel_baru_3 = aselole[['Partner', 'Dependents']]
tabel_baru_3.reset_index(drop=True, inplace=True)
#tabel_baru_3

len(tabel_baru_3)
#jumlahnya 359

359

In [35]:
#soal 5

con_total = df_edit['TotalCharges']
con_senior = df_edit['SeniorCitizen'] == 0
con_phone = df_edit['PhoneService'] == 'Yes' 
con_Internet = df_edit['InternetService'] == 'Fiber optic'


final_Q = df_edit[con_senior & con_phone & con_Internet & con_total].copy() #int(con_total) harus diakhir kalau ingin menghindari error

final_Q[['SeniorCitizen', 'PhoneService', 'InternetService', 'TotalCharges']]
final_Q


#final_Q['TotalCharges'] = pd.to_numeric(final_Q['TotalCharges'], errors='coerce') #mengubah str number menjadi numeric number


#tabel_final.applymap(np.isreal) #mengecek data total charges apakah ada yg string atau tidak



#mean_tabel_final = final_Q['TotalCharges'].mean()
#print('rata rata total charges:', mean_tabel_final)

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
6     1452-KIOVK    Male              0      No        Yes      22   
8     7892-POOKP  Female              0     Yes         No      28   
12    8091-TTVAX    Male              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
7033  9767-FFLEM    Male              0      No         No      38   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
4             Yes            No     Fiber optic             No  ...   
5             Yes           Yes     Fiber optic             No  ...   
6             Yes           Yes     Fiber optic             No  ...   
8             Yes           Yes     Fiber optic             No  ...   
12            Yes           Yes     Fiber optic             No  ...   
...           ...           ...             ...            ...  ...   
7033          Yes            No     Fiber optic             No  ...   
7034          Yes           Yes     Fiber optic            Yes  ...   
7035          Yes            No     Fiber optic             No  ...   
7039          Yes           Yes     Fiber optic             No  ...   
7042          Yes            No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
4                  No          No          No              No  Month-to-month   
5                 Yes          No         Yes             Yes  Month-to-month   
6                  No          No         Yes              No  Month-to-month   
8                 Yes         Yes         Yes             Yes  Month-to-month   
12                Yes          No         Yes             Yes        One year   
...               ...         ...         ...             ...             ...   
7033               No          No          No              No  Month-to-month   
7034              Yes          No         Yes              No  Month-to-month   
7035               No          No         Yes              No  Month-to-month   
7039              Yes          No         Yes             Yes        One year   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
4                 Yes           Electronic check          70.70        151.65   
5                 Yes           Electronic check          99.65         820.5   
6                 Yes    Credit card (automatic)          89.10        1949.4   
8                 Yes           Electronic check         104.80       3046.05   
12                 No    Credit card (automatic)         100.35        5681.1   
...               ...                        ...            ...           ...   
7033              Yes    Credit card (automatic)          69.50       2625.25   
7034              Yes    Credit card (automatic)         102.95       6886.25   
7035              Yes  Bank transfer (automatic)          78.70        1495.1   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
4      Yes  
5      Yes  
6       No  
8      Yes  
12      No  
...    ...  
7033    No  
7034   Yes  
7035    No  
7039    No  
7042    No  

[2265 rows x 21 columns]

### Silakan menjawab di bawah ini:

#### Soal 1:
Buanglah baris dengan tenure samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan tenure TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

#### Jawaban: 7032 (langsung edit di sini aja)

#### Soal 2:
Bagi customer yang TIDAK MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa rata-rata dari tenure mereka?

#### Jawaban: 20.374376039933445

#### Soal 3:
Bagi customer yang MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa maksimal MonthlyCharges yang tercata?

#### Jawaban: 118.6

#### Soal 4:
Berapa banyak customer yang TIDAK MEMILIKI PARTNER namun MEMILIKI DEPENDENTS?

#### Jawaban: 359

#### Soal 5:
Berapa rata-rata TOTAL CHARGES dari customer NON-SENIOR CITIZEN (Senior Citizen = 0) yang memiliki PHONE SERVICE dan memiliki INTERNET SERVICE FIBER OPTIC?

#### Jawaban: 3226.4927373068435